app/assets/src/components/views/discovery/DiscoveryView.jsx

```diff
@@ -547,7 +547,9 @@ class DiscoveryView extends React.Component {
         break;
       }
       case "project": {
-        this.handleProjectSelected({ project: this.projects.get(value) });
+        this.handleProjectSelected({
```

Comment:
I don't understand this part of the code, but this is confusing in isolation.

Here, it's okay to just set a "placeholder" project without any fields besides id? That doesn't lead to any errors where another component is expecting project to have more fields?

Maybe add a comment explaining why it's okay...

Reply:
Yes, it tries to load the whole project if possible to avoid requesting from the server and accelerate rendering.
If not send the id only and the project will be loaded from server.